src/components/hooks/HookDetails.tsx

```diff
@@ -0,0 +1,185 @@
+      targetState = hookInput[0].payload.data.task.status;
+      break;
+    case 'on_build':
+      targetName = 'Build';
+      targetState = hookInput[0].payload.data.build.status;
+      break;
+    default:
+      targetName = 'Unknown';
+      targetState = 'UNKNOWN';
+      break;
+  }
+
+  // In case this is a task hook — display a chip that links to it
+  const taskChip =
+    hook.task != null ? <TaskNameChip className={classes.chip} task={hook.task} withNavigation={true} /> : null;
+
+  // Potential error from the hook execution
+  let headerStyle = {
+    backgroundColor: useNotificationColor('ERROR'),
+  };
+  const potentialError =
+    hook.info.error === '' ? null : (
+      <div className={classNames('container', classes.gap)}>
+        <div
+          key={props.hook.info.error}
+          style={headerStyle}
+          className={classNames('row', 'justify-content-between', 'align-items-center')}
+        >
+          <Typography variant="subtitle1" className={classes.potentialError}>
+            {props.hook.info.error}
+          </Typography>
+        </div>
+      </div>
+    );
+
+  return (
+    <div>
+      <CirrusFavicon status={hook.info.error === ''} />
+      <Head>
+        <title>{targetName} hook - Cirrus CI</title>
+      </Head>
+      <Card>
+        <CardContent>
+          <div className={classes.wrapper}>
+            <RepositoryNameChip className={classes.chip} repository={hook.repository} />
+            <BuildBranchNameChip className={classes.chip} build={hook.build} />
+            <BuildChangeChip className={classes.chip} build={hook.build} />
+            {taskChip}
+          </div>
+          <div className={classes.wrapper}>
+            <HookCreatedChip className={classes.chip} hook={hook} />
+            <HookStatusChip className={classes.chip} hook={hook} />
+          </div>
+          <div className={classes.gap} />
+          <Typography variant="h6" gutterBottom>
+            {targetName} hook
+          </Typography>
+          <p>
+            Triggered because <code>{hook.name}()</code> was defined and the has task entered the{' '}
+            <code>{targetState}</code> state.
+          </p>
+        </CardContent>
+      </Card>
+      {potentialError}
+      <div className={classes.gap} />
+      <Card>
+        <CardContent>
+          <Typography variant="h6">Execution logs</Typography>
+          <Logs logsName="output" logs={props.hook.info.outputLogs.join('\n')} />
+        </CardContent>
+      </Card>
+      <div className={classes.gap} />
+      <Card>
+        <CardContent>
+          <Typography variant="h6">Input</Typography>
+          <pre className={classes.io}>{prettyHookInput}</pre>
+        </CardContent>
+      </Card>
+      <div className={classes.gap} />
+      <Card>
+        <CardContent>
+          <Typography variant="h6">Output</Typography>
+          <pre className={classes.io}>{prettyHookOutput}</pre>
+        </CardContent>
+      </Card>
```

Comment:
Let's not show the output if there is none. At the moment hooks can return some values but they are not used.

Reply:
See 1326e7c1796883cb0c76b4986efd220d44c822c9.